twilio/endpoint.bal

```diff
@@ -36,14 +36,24 @@ public isolated client class Client {
     # + twilioConfig - Twilio connection configuration 
     # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized
     public isolated function init(ConnectionConfig twilioConfig, http:ClientConfiguration httpClientConfig = {}) returns error? {
-        self.accountSId = twilioConfig.accountSId;
-        auth:CredentialsConfig authConfig = {
-            username: twilioConfig.accountSId,
-            password: twilioConfig.authToken
-        };
-        http:ClientConfiguration httpConfig = httpClientConfig;
-        httpConfig.auth = authConfig;
-        self.basicClient = check new (TWILIO_API_BASE_URL, httpConfig);
+        self.accountSId = twilioConfig.authConfig.accountSId;
+        if (twilioConfig.authConfig is TokenBasedAuthentication) {
+            auth:CredentialsConfig authConfig = {
```

Comment:
We can declare this `authConfig ` before the `if else` block with a common type.
Later, within the `if else` block, we can check the type of the record and populate the values.

By doing this, we can have the 
```ballerina
http:ClientConfiguration httpConfig = httpClientConfig;
httpConfig.auth = authConfig;
self.basicClient = check new (TWILIO_API_BASE_URL, httpConfig);
```
after the `if else` block. No need to add this 2 times as Roland said.